### PR TITLE
add pytest benchmark

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR adds a `pytest-benchmark` suite to follow the performance of the `items` endpoint and an additional GitHub action that will compare the benchmark results and comment if needed


```
python -m pytest --benchmark-only --benchmark-columns 'min, max, mean, median'                                             
==================================================================================================================================== test session starts =====================================================================================================================================
platform darwin -- Python 3.8.9, pytest-7.0.1, pluggy-1.0.0
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/vincentsarago/Dev/DevSeed/tifeatures
plugins: pgsql-1.1.2, benchmark-3.4.1, anyio-3.5.0, asyncio-0.17.2, Faker-13.3.1, profiling-1.7.0, hypothesis-6.46.11, cov-3.0.0
asyncio: mode=legacy
collected 41 items                                                                                                                                                                                                                                                                           

tests/test_benchmarks.py ..................                                                                                                                                                                                                                                                                                                                              [ 43%]
tests/test_main.py s                                                                                                                                                                                                                                                                                                                                                     [ 46%]
tests/routes/test_collections.py sss                                                                                                                                                                                                                                                                                                                                     [ 53%]
tests/routes/test_endpoints.py sss                                                                                                                                                                                                                                                                                                                                       [ 60%]
tests/routes/test_item.py s                                                                                                                                                                                                                                                                                                                                              [ 63%]
tests/routes/test_items.py ssssssssssssss                                                                                                                                                                                                                                                                                                                                [ 97%]
tests/routes/test_templates.py s                                                                                                                                                                                                                                                                                                                                         [100%]

------------------------------------------ benchmark 'csv': 6 tests ------------------------------------------
Name (time in ms)                     Min                 Max                Mean             Median          
--------------------------------------------------------------------------------------------------------------
test_benchmark_items[csv-1]        5.1540 (1.0)        6.2745 (1.0)        5.6902 (1.0)       5.6814 (1.0)    
test_benchmark_items[csv-10]       8.9532 (1.74)      25.2055 (4.02)      10.9825 (1.93)     10.1417 (1.79)   
test_benchmark_items[csv-50]      22.8745 (4.44)      27.2879 (4.35)      24.0670 (4.23)     23.8828 (4.20)   
test_benchmark_items[csv-100]     40.3293 (7.82)      42.1151 (6.71)      41.1513 (7.23)     41.1011 (7.23)   
test_benchmark_items[csv-200]     77.7404 (15.08)     81.9049 (13.05)     79.5599 (13.98)    79.1462 (13.93)  
test_benchmark_items[csv-250]     96.4666 (18.72)    130.3855 (20.78)    102.5506 (18.02)    99.9151 (17.59)  
--------------------------------------------------------------------------------------------------------------

------------------------------------------- benchmark 'geojson': 6 tests -------------------------------------------
Name (time in ms)                          Min                 Max                Mean              Median          
--------------------------------------------------------------------------------------------------------------------
test_benchmark_items[geojson-1]         6.3323 (1.0)        7.1660 (1.0)        6.5611 (1.0)        6.5129 (1.0)    
test_benchmark_items[geojson-10]       15.3395 (2.42)      17.9609 (2.51)      16.4820 (2.51)      16.5790 (2.55)   
test_benchmark_items[geojson-50]       49.8940 (7.88)      90.5500 (12.64)     56.4390 (8.60)      51.7277 (7.94)   
test_benchmark_items[geojson-100]      94.3629 (14.90)    119.7988 (16.72)     98.6869 (15.04)     96.0438 (14.75)  
test_benchmark_items[geojson-200]     188.0974 (29.70)    214.3958 (29.92)    197.3848 (30.08)    189.5700 (29.11)  
test_benchmark_items[geojson-250]     234.8502 (37.09)    261.9789 (36.56)    245.6660 (37.44)    234.9135 (36.07)  
--------------------------------------------------------------------------------------------------------------------

------------------------------------------- benchmark 'html': 6 tests -------------------------------------------
Name (time in ms)                       Min                 Max                Mean              Median          
-----------------------------------------------------------------------------------------------------------------
test_benchmark_items[html-1]         6.1564 (1.0)       18.6733 (1.0)        7.6271 (1.0)        6.8630 (1.0)    
test_benchmark_items[html-10]       13.0805 (2.12)      40.0632 (2.15)      20.3001 (2.66)      18.5978 (2.71)   
test_benchmark_items[html-50]       39.0653 (6.35)      67.6766 (3.62)      41.5760 (5.45)      40.2365 (5.86)   
test_benchmark_items[html-100]      72.4877 (11.77)    111.9789 (6.00)      80.0676 (10.50)     76.1592 (11.10)  
test_benchmark_items[html-200]     142.3825 (23.13)    175.6181 (9.40)     152.8848 (20.05)    144.7540 (21.09)  
test_benchmark_items[html-250]     177.5379 (28.84)    209.4210 (11.22)    189.3855 (24.83)    180.9182 (26.36)  
-----------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
================================================================================================================================================================= 18 passed, 23 skipped, 6 warnings in 24.76s ==================================================================================================================================================================
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -63,6 +63,34 @@ jobs:
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install .["all,test"]
+
+      - name: Run Benchmark
+        run: python -m pytest --benchmark-only --benchmark-autosave --benchmark-columns 'min, max, mean, median' --benchmark-json output.json
+
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:85]: never used this, let see how it goes!

Comment (vincentsarago) [reply .github/workflows/ci.yml:93]: `github-action-benchmark` is a nice gh action (https://github.com/benchmark-action/github-action-benchmark) which can create HTML page to follow the evolution of the performance see : https://vincentsarago.github.io/lebanc/benchmarks/

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -63,6 +63,47 @@ jobs:
...
+      - name: Run Benchmark
+        run: python -m pytest --benchmark-only --benchmark-columns 'min, max, mean, median' --benchmark-json output.json
+
+      - name: Store and benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: TiFeatures Benchmarks
+          tool: 'pytest'
+          output-file-path: output.json
+          alert-threshold: '130%'
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:98]: that could be loosen a bit

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -63,6 +63,47 @@ jobs:
...
+          comment-on-alert: true
+          fail-on-alert: true
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:100]: We fail and alert if the performance is bad!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -63,6 +63,47 @@ jobs:
...
+          # GitHub API token to make a commit comment
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Make a commit on `gh-pages` only if master
+          auto-push: ${{ github.ref == 'refs/heads/master' }}
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:104]: we only push (create the HTML page) when in `master` branch

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -63,6 +63,47 @@ jobs:
...
+          benchmark-data-dir-path: benchmarks
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:105]: we add benchmarks in the `gh-pages` branch within the `benchmarks/` directory

**Comment by vincentsarago** [comment]
@bitner let me know if you have question 🙏

**Review by bitner — APPROVED** [review]

**Pull request merged**